sklearn/kernel_approximation.py

```diff
@@ -682,8 +687,36 @@ def fit(self, X, y=None):
         X = self._validate_data(X, accept_sparse="csr")
         check_non_negative(X, "X in AdditiveChi2Sampler.fit")
 
+        if self.sample_interval is None:
+            # See figure 2 c) of "Efficient additive kernels via explicit feature maps"
+            # <http://www.robots.ox.ac.uk/~vedaldi/assets/pubs/vedaldi11efficient.pdf>
+            # A. Vedaldi and A. Zisserman, Pattern Analysis and Machine Intelligence,
+            # 2011
+            if self.sample_steps == 1:
+                self._sample_interval = 0.8
+            elif self.sample_steps == 2:
+                self._sample_interval = 0.5
+            elif self.sample_steps == 3:
+                self._sample_interval = 0.4
+            else:
+                raise ValueError(
+                    "If sample_steps is not in [1, 2, 3],"
+                    " you need to provide sample_interval"
+                )
+        else:
+            self._sample_interval = self.sample_interval
```

Comment:
We cannot do that. This is mutating the attribute passed by the user.
We should keep `self.sample_interval` as given in `__init__`

Reply:
So, using `copy()`?

Reply:
Actually, I don't understand this feedback. The logic of `fit` used to be:
 ```python
 self.sample_interval_ = self.sample_interval
```
so this was mutating the user's attribute as well, correct?